odoo/models.py

```diff
@@ -3121,25 +3121,14 @@ def fields_get(self, allfields=None, attributes=None):
         :param allfields: list of fields to document, all if empty or not provided
         :param attributes: list of description attributes to return for each field, all if empty or not provided
         """
-        has_access = functools.partial(self.check_access_rights, raise_exception=False)
-        readonly = not (has_access('write') or has_access('create'))
-
```

Comment:
I may be wrong, but I think this change breaks existing behavior.  When exporting data in import-compatible way, readonly fields are discarded.  I don't know whether this was intended on models that the user cannot update, but it is a change.

Reply:
I will have a look into that

Reply:
Yes, there is indeed a change of behavior:

With the demo user, attempting to export pricelists, to which the user has only the read access

Before:
![image](https://user-images.githubusercontent.com/5822488/161036435-dcf9a739-78b4-4373-bb4c-6410be09643b.png)


After:
![image](https://user-images.githubusercontent.com/5822488/161036499-c617b9b2-bbae-4baf-befe-3d131f1e5686.png)

That being said, I do not think this is a bad behavior: If the goal of the user is for example to export the pricelists in another database to which he has the create/write access to pricelists, it makes sense.

If the user intended goal was to import back in the same database in which he doesn't have the access rights, well first he doesn't see the "import records" button, and even if he did, he would have received an error he doesn't have the access rights to edit the pricelists, which is a more meaningful message to understand the reason he cannot import back the file, rather than an empty fields list in the export wizard *-).
Having the field list empty in the export wizard when the user doesn't have the create/edit access to the model is simply not meaningful for the user in my opinion. He doesn't have the reason *why* the list is empty.

Reply:
I agree with you, I also find this behavior change reasonable.